File: packages/room-server/src/entities/datasheet.record.entity.ts
import { Column, Entity } from 'typeorm';
import { BaseEntity } from './base.entity';
import { IRecordCellValue, IRecordMeta } from '@vikadata/core';

/**
 * 工作台-数表记录表
 */
@Entity('vika_datasheet_record')
export class DatasheetRecordEntity extends BaseEntity {
  @Column({
    name: 'record_id',
    nullable: true,
    comment: '操作ID',
    length: 50,
  })
    recordId: string | null;

  @Column({
    name: 'dst_id',
    nullable: true,
    comment: '数表ID(关联#vika_datasheet#dst_id)',
    length: 50,
  })
    dstId: string | null;

  @Column('json', {
    name: 'data',
    nullable: true,
    comment: '一行记录的数据（对应每个字段）',
  })
    data: IRecordCellValue | null;

  @Column({
    name: 'revision_history',
    nullable: true,
    comment: '按排序的历史版本号，是原 Operation 的revision，数组下标是当前 record 的 revision',
    length: 5000,
    default: () => '0',
  })
    revisionHistory: string | null;

  @Column({
    name: 'revision',
    nullable: true,
    comment: '版本号',
    unsigned: true,
    default: () => '0',
  })
    revision: number | null;

  @Column('json', {
    name: 'field_updated_info',
    nullable: true,
    comment: '单元格更新信息',
  })
    recordMeta?: IRecordMeta;
}

File: backend-server/db-model/src/main/java/com/vikadata/entity/DatasheetRecordEntity.java
package com.vikadata.entity;

import java.io.Serializable;
import java.time.LocalDateTime;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * 工作台-数表记录表
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "datasheet_record")
public class DatasheetRecordEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 主键
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 操作ID
     */
    private String recordId;

    /**
     * 数表ID(关联#vika_datasheet#dst_id)
     */
    private String dstId;

    /**
     * 一行记录的数据（对应每个字段）
     */
    private String data;

    /**
     * 按排序的历史版本号，是原 Operation 的revision，数组下标是当前 record 的 revision
     */
    private String revisionHistory;

    /**
     * 版本号
     */
    private Long revision;

    /**
     * 字段更新信息
     */
    private String fieldUpdatedInfo;

    /**
     * 删除标记(0:否,1:是)
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * 创建用户
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * 最后一次更新用户
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;

}
